beacon-chain/cache/depositsnapshot/finalized_node.go

```diff
@@ -0,0 +1,34 @@
+package depositsnapshot
+
+import "github.com/pkg/errors"
+
+// Finalized represents a finalized node and satisfies the MerkleTreeNode interface.
+type Finalized struct {
+	deposits uint64
+	hash     [32]byte
+}
```

Comment:
Want to name this struct and others with the `Node` suffix like `FinalizedNode` ?

Reply:
Yes, that will be much clearer